course7/homeworks/fred/complex-rpn/complex.h

```diff
@@ -0,0 +1,21 @@
+#pragma once
+
+struct Complex
+{
+  double real;
+  double imaginary;
+
+  Complex operator+ (const Complex& other) const;
+  Complex operator- (const Complex& other) const;
+  Complex operator* (const Complex& other) const;
+  Complex operator/ (const Complex& other) const;
+};
+
+template <typename Stream>
+Stream& operator<< (Stream& stream, const Complex& c) {
```

Comment:
This should be the other way around:
- the ostream operator, which is not perf critical and is rather large, should be in the .cc file.
- the small arithmetical operations, that would benefit from inlining, should be defined here in the header.

Reply:
What if I add the word 'inline' to the arithmetical operations definitions? Would that have the same effect as putting the implementation in the header file?

Reply:
It won't help at all. To inline the function, compiler has to access the function's definition when compiling calculator.cc, so it has to be in the included header.